### PR TITLE
Split long literals with underscores

### PR DESCRIPTION
**Pull request by ljedrz**
This reduces the amount of clippy warnings for the compiler by around 1MB.

### CODE REVIEW

**Comment by varkor** [comment]
The largest changes here are to `src/libcore/num/dec2flt/table.rs` and `src/libcore/unicode/tables.rs`, which are generated by `src/etc/dec2flt_table.py` and `src/libcore/unicode/unicode.py` respectively. Rather than reformatting the files directly, the generating scripts should be updated so that the next time they're run, it won't reset these formatting changes.

**Comment by Mark-Simulacrum** [comment]
Most of these changes don't really seem to be actually helpful - changing an array of constants or specific hash values to be underscore separated doesn't really help; no one is actually going to be reading them anyway. So personally I'm against a blanket change like this.

File: src/libcore/num/f32.rs [hunk 1]
```diff
@@ -80,75 +80,75 @@ pub mod consts {
 
     /// Archimedes' constant (π)
     #[stable(feature = "rust1", since = "1.0.0")]
-    pub const PI: f32 = 3.14159265358979323846264338327950288_f32;
+    pub const PI: f32 = 3.141_592_653_589_793_238_462_643_383_279_502_88_f32;
```
Comment (varkor) [inline src/libcore/num/f32.rs]: It's arguable whether this is actually an improvement. If anyone wants to double-check this constant, they'll almost certainly be comparing against a number _without_ underscores, in which case the underscores get in the way. I doubt anyone will be directly reading the literals here either.

**Comment by varkor** [comment]
If the goal for this change is to reduce clippy warnings, perhaps we can override / improve the warnings instead of making these unnecessary formatting changes.

**Comment by ljedrz** [comment]
I'm ok with either decision; I didn't modify those by hand 😌.

**Comment by Mark-Simulacrum** [comment]
Let's close this then; I don't think we should land anything that has allows for clippy lints in rust-lang/rust and we seem to mostly agree that this isn't actually a desired change.

**Pull request closed without merging**